### PR TITLE
[microTVM][RVM] Skip USB device attach if device is already attached

### PR DESCRIPTION
**Pull request by gromero**
Currently - when the VirtualBox provider is used - if base-box-tool.py
'test' command is used and a VM is already running with the USB device
used to perform the tests already attached to it the command fails
because it tries to blindly attach again the USB device without checking
if it is already attached.

This commit fixes that error by checking and properly skipping the USB
device if it's already attached to the VirtualBox VM.

Signed-off-by: Gustavo Romero <gustavo.romero@linaro.org>

Thanks for contributing to TVM!   Please refer to guideline https://tvm.apache.org/docs/contribute/ for useful information and tips. After the pull request is submitted, please request code reviews from [Reviewers](https://github.com/apache/incubator-tvm/blob/master/CONTRIBUTORS.md#reviewers) by @ them in the pull request thread.

### CODE REVIEW

**Comment by gromero** [comment]
@mehrdadh could you please take a look? I think I hit on a issue similar to what you found when you added the TODO(). Thank you.

**Comment by gromero** [comment]
Retriggering since https://github.com/apache/tvm/pull/8927 workaround is in place.

**Review by mehrdadh — APPROVED** [review]
@gromero thanks for fixing this issue.
LGTM!

**Comment by gromero** [comment]
> @gromero thanks for fixing this issue.
> LGTM!

@mehrdadh Thanks for the review!

**Comment by gromero** [comment]
@areusch I'm wondering if you could bless it : )

File: apps/microtvm/reference-vm/base-box-tool.py [hunk 1]
```diff
@@ -74,10 +74,31 @@ def parse_virtualbox_devices():
     return devices
 
 
+VIRTUALBOX_USB_DEVICE_RE = (
+    "UUID:.*([0-9a-z-]{36}).*\nVendorId:.*\(([0-9A-Z]{4})\).*\nProductId:.*\(([0-9A-Z]{4})\)\n"
+)
+
+
+def parse_virtualbox_attached_usb_devices(vm_uuid):
+    output = subprocess.check_output(["VBoxManage", "showvminfo", vm_uuid], encoding="utf-8")
```
Comment (areusch) [inline apps/microtvm/reference-vm/base-box-tool.py:83]: if you use `--machinereadable` here, it should be a bit easier to parse and more robust to updates in VBoxManage. would you be up for trying that?

**Review by areusch — COMMENTED** [review]
hi @gromero sorry i had a bit of a problem with my GH notifications and didn't see this one. left one question for you here.

**Comment by areusch** [comment]
@gromero could you address my comment above?

**Comment by gromero** [comment]
> @gromero could you address my comment above?

@areusch sorry for the delay updating it. PTAL. Thanks!

**Review by areusch — APPROVED** [review]
thanks @gromero!

**Review by leandron — APPROVED** [review]
LGTM, thanks @gromero

**Pull request merged**